src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -563,7 +563,19 @@ namespace winrt::TerminalApp::implementation
     void TerminalPage::_UpdateTitle(std::shared_ptr<Tab> tab)
     {
         auto newTabTitle = tab->GetFocusedTitle();
-        tab->SetTabText(newTabTitle);
+        const auto lastFocusedProfile = tab->GetFocusedProfile().value();
+
+        auto matchingProfile = _settings->FindProfile(lastFocusedProfile);
+        if (matchingProfile)
+        {
+            if (matchingProfile->GetSuppressApplicationTitle())
+            {
+                auto profileTabTitle = matchingProfile->GetTabTitle();
+                newTabTitle = profileTabTitle.empty() ? matchingProfile->GetName() : profileTabTitle;
```

Comment:
Why does this operate on this side of the equation instead of preventing tab->SetFocusedTitle() from storing a client application set title in the first place?